### PR TITLE
US99396 Add use-enrollment-cards LD flag

### PR DESCRIPTION
**Pull request by ryantmer**
This flag will cut between the two possible representations of courses - with the flag on, we use the new (but not yet complete) `d2l-enrollment-card`. When it's off (current default), we use the old `d2l-course-imate-tile`. Once we are ready to release the `d2l-enrollment-card` and for it to entirely replace the tile, we can remove this flag.

This is _kind of_ the opposite of https://github.com/Brightspace/d2l-my-courses-ui/pull/574, with a few more changes here and there to make things work. Ultimately though, it's just re-adding a reference to the `1.x-hybrid-components` repo, and making things `useEnrollmentCards`-LD-flag-aware. Also used [this diff](https://github.com/Brightspace/d2l-my-courses-ui/compare/a1e8ed865090df024589471eb8147768c9d21539...master) as a reference, which is the diff between master and when the `d2l-course-image-tile` was removed.

### CODE REVIEW

File: src/d2l-my-courses-content/d2l-my-courses-content-behavior.html [hunk 1]
```diff
@@ -532,6 +545,10 @@
 			allCourses.updatedSortLogic = this.updatedSortLogic;
 			allCourses.cssGridView = this.cssGridView;
 			allCourses.presentationUrl = this.presentationUrl;
+			allCourses.showCourseCode = this.showCourseCode;
+			allCourses.showSemester = this.showSemester;
+			allCourses.courseUpdatesConfig = this.courseUpdatesConfig;
```
Comment (m6jones) [inline src/d2l-my-courses-content/d2l-my-courses-content-behavior.html]: So these don't have properties defined. Is that a problem? Still work and would be easier to cleanup?

**Comment by ryantmer** [comment]
I'll take a look at the pinning! Thought I had checked that, but there are manymanymany different edge cases and such 😰

File: src/d2l-all-courses-unified-content.html [hunk 2]
```diff
@@ -36,13 +37,28 @@
...
-				<div>
-					<d2l-enrollment-card
-						href="[[item]]",
-						presentation-href=[[presentationUrl]]>
-					</d2l-enrollment-card>
-				</div>
+			<template is="dom-if" if="[[useEnrollmentCards]]">
+				<template is="dom-repeat" items="[[filteredEnrollments]]">
+					<div>
+						<d2l-enrollment-card
```
Comment (njostonehouse) [inline src/d2l-all-courses-unified-content.html]: Shouldn't the `dom-if` be inside the `dom-repeat`?

Comment (m6jones) [reply src/d2l-all-courses-unified-content.html]: If it was inside the dom repeat then it would be called a lot more? Isn't that bad?

Comment (njostonehouse) [reply src/d2l-all-courses-unified-content.html]: Hm, I don't know. I would've assumed it would only get it once, but you may be right.

**Comment by m6jones** [comment]
Course Code and Semester names were not showing up on my local build.

File: src/d2l-all-courses-unified-content.html (hunk 2 above)
Comment (ryantmer) [reply src/d2l-all-courses-unified-content.html]: I'm actually not sure either - I _think_ what would happen the other way is you'd end up with a set of `n` enrollment cards (or tiles), and a set of `n` empty `template`s!

**Comment by ryantmer** [comment]
To confirm, the pinning/course codes/semester not showing is when the flag is off (and you're seeing `d2l-course-image-tile`s), correct?

File: src/d2l-all-courses-unified-content.html (hunk 2 above)
Comment (njostonehouse) [reply src/d2l-all-courses-unified-content.html]: That's true, it's probably best to keep the "hidden" templates to a minimum too. Cool, I'm sold.

**Comment by m6jones** [comment]
Yes, I am not seeing it for the d2l-course-image-tile.

**Comment by ryantmer** [comment]
Gotcha. I'll dig into it - fortunately course code/semester was just me forgetting to pass those attributes down... whoops.

**Comment by ryantmer** [comment]
Updated! Fortunately, small fixes for both problems. 😌

**Comment by m6jones** [comment]
All Courses view is dropping the Course Code and Semester. Otherwise the update is good!

**Comment by m6jones** [comment]
Note that updates are not showing up on both views. (All Courses and the My Courses)

**Pull request merged**